library/extractor/src/main/java/com/google/android/exoplayer2/text/ssa/SsaDecoder.java

```diff
@@ -376,11 +348,77 @@ private static Cue createCue(
       cue.setPosition(styleOverrides.position.x / screenWidth);
       cue.setLine(styleOverrides.position.y / screenHeight, LINE_TYPE_FRACTION);
     } else {
-      // TODO: Read the MarginL, MarginR and MarginV values from the Style & Dialogue lines.
       cue.setPosition(computeDefaultLineOrPosition(cue.getPositionAnchor()));
       cue.setLine(computeDefaultLineOrPosition(cue.getLineAnchor()), LINE_TYPE_FRACTION);
     }
 
+    // Apply margins if there are no overrides and we have valid positions.
+    if (styleOverrides.alignment == SsaStyle.SSA_ALIGNMENT_UNKNOWN
+        && styleOverrides.position == null
+        && cue.getPosition() != Cue.DIMEN_UNSET
+        && cue.getLine() != Cue.DIMEN_UNSET) {
+      // Margin from Dialogue lines takes precedence over margin from Style line.
+      cue.setPosition(cue.getPosition() + (marginLeft != 0f
```

Comment:
I could be wrong, but it looks like if a user sets the margin via a style then they can't override it to `0` in the dialogue line, because `0` is interpreted to mean `unset` and therefore ignored?

Possibly we need to represent this slightly differently: Store intermediate `unset` values, so we can correctly decide whether to take the value from the style or dialogue line - and if both are unset then only then default to `0`.

EDIT: Actually looking below it seems you're already using `Cue.DIMEN_UNSET`, so maybe this line should just be updated to check for `DIMEN_UNSET` rather than `0`?

Reply:
Overriding a margin to `0` with dialogue lines it's not a thing - based on my tests with VLC and also based on the documentation: 

> MarginL   4-figure Left Margin override. The values are in pixels. All zeroes means the default margins defined by the style are used.

Reply:
Agreed, thanks - I think we can probably remove all references to `DIMEN_UNSET` margin values. I'll do that when merging internally.